cmd/delete/cmd.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package delete
+
+import (
+	"context"
+	"errors"
+	"github.com/minio/minio-go/pkg/s3utils"
+	"github.com/spf13/cobra"
+
+	minio "github.com/minio/minio/cmd"
+	cmdUtils "storj.io/ditto/cmd/utils"
+)
+
+var mirroring = cmdUtils.GetObjectLayer
+
+var Cmd = &cobra.Command{
+	Use:   "delete ",
+	Args:  validateArgs,
+	Short: "Delete files and buckets.",
+	Long: "to delete object - delete bucketName objectName.\n" +
+		"to delete bucket - delete bucketName.\n",
+	RunE: exec,
+}
+
+//Error messages ares
+var (
+	noArgsMessage 			 = "at least one argument required"
+	prefixMissingMessage 	 = "prefix argument is missing"
+	bucketNameInvalidMessage = "bucket name is not valid"
+)
+
+//Flags values area
+var (
+	forceFlag     bool
+	recursiveFlag bool
+	prefixFlag    bool
+	delimeterFlag string
+)
+
+func exec(cmd *cobra.Command, args []string) (err error) {
+
+	var bucketName, objectName, prefix, delimiter string
+	bucketName = args[0]
+	delimiter = "/"
+	var ctx = context.Background()
+
+	p  := prefixFlag
+	rf := recursiveFlag && forceFlag
+	noFlags := !(recursiveFlag && prefixFlag && forceFlag)
+
+	if p {
+		objectName, prefix = args[0], args[1]
+	}
+
+	m, err := mirroring()
+	if err != nil {
+		return err
+	}
+
+	switch {
+		case prefixFlag:
+
+			return deleteRecursive(ctx, m, bucketName, prefix, delimiter)
+
+		case rf:
```

Comment:
Maybe we should get rid of this boolean flags and move it declaration to case statements?